### PR TITLE
ClassDefinitionFixer - Anonymous classes format by PSR12

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2336
master version of https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2328

Fixes
- anonymous classes definitions
- edge case with `#` comment within the definition

Please review @gharlan (let me know if cases are missing)

### CODE REVIEW

File: tests/Fixer/ClassNotation/ClassDefinitionFixerTest.php [hunk 1]
```diff
@@ -41,16 +43,17 @@ public function testConfigureDefaultToNull()
     }
 
     /**
-     * @param string $expected PHP source code
-     * @param string $input    PHP source code
+     * @param string     $expected PHP source code
+     * @param string     $input    PHP source code
+     * @param null|mixed $config
```
Comment (keradus) [inline tests/Fixer/ClassNotation/ClassDefinitionFixerTest.php]: `null|array` ?

Comment (SpacePossum) [inline tests/Fixer/ClassNotation/ClassDefinitionFixerTest.php]: ah yes, PHPStorm guessed wrong ;)

**Comment by GrahamCampbell** [comment]
Anonymous classes need to be in the configuration, otherwise, we'll have to remove this from the psr ruleset.

**Comment by keradus** [comment]
they are in the spirit of psr-2 anonymous methods.
the thing is that when document was created, anonymous classes was not possible to made

**Comment by GrahamCampbell** [comment]
Right, but the psr-2 spec officially doesn't enforce anything, only psr-12, right?

**Comment by SpacePossum** [comment]
>Anonymous classes need to be in the configuration, otherwise, we'll have to remove this from the psr2 ruleset.

I vote to remove it from the PSR2 set than as imo there is no point in splitting up the fixer.

**Comment by keradus** [comment]
almost nobody uses psr-12

**Comment by SpacePossum** [comment]
I don't get it, the was done on 1.x and merge, than requested to do on master and now it is a problem?

**Comment by keradus** [comment]
i don't have problem with it, I am waiting for @gharlan check as you asked him ;)

**Comment by keradus** [comment]
Thank you @SpacePossum.
